planner/core/task.go

```diff
@@ -2345,17 +2363,20 @@ func (t *mppTask) convertToRootTaskImpl(ctx sessionctx.Context) *rootTask {
 	return rt
 }
 
-func (t *mppTask) needEnforce(prop *property.PhysicalProperty) bool {
+func (t *mppTask) needEnforceExchanger(prop *property.PhysicalProperty) bool {
 	switch prop.MPPPartitionTp {
 	case property.AnyType:
 		return false
 	case property.BroadcastType:
 		return true
+	case property.PassThroughType:
+		return t.partTp != property.PassThroughType
 	default:
 		if t.partTp != property.HashType {
 			return true
 		}
 		// TODO: consider equalivant class
+		// TODO: `prop.IsSubsetOf` is enough, instead of equal.
```

Comment:
not really, depend on the hash functions. for example hash(A,B,C) guarantee the same B,C distribute on a same node?

Reply:
I mean the opposite, for example, already partitioned by hash(B,C), then same (A,B,C) must distribute on a same node.

Reply:
added more comments